.github/workflows/pr_and_main.yml

```diff
@@ -0,0 +1,61 @@
+name: PR and main CI
+
+on: 
+  schedule:
+    - cron: '0 1 * * *'
+  pull_request:
+  
+
+jobs:
+  build_and_test:
+    name: CI (${{ matrix.os }}, ${{ matrix.python-version }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        python-version: ["3.7"]
+        include:
+        - os: ubuntu-latest
+          cmake-preset: ci-linux
+        - os: macos-latest
+          cmake-preset: ci-macos
+        - os: windows-latest
+          cmake-preset: ci-windows
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+          fetch-depth: 0  # history required so cmake can determine version
+
+      - name: Setup env variables
+        run: |
+          echo "PYTHONPATH=$(pwd)/install" >> $GITHUB_ENV
+
+      - name: ccache
+        if: ${{ !contains(matrix.os, 'windows') }}
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: ${{ matrix.os }}
+          max-size: 1000M
+
+      - name: Setup Mambaforge and dev env
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-version: latest
+          miniforge-variant: Mambaforge
```

Comment:
Does this mean that we don't have to set up (micro)mamba ourselves?

Reply:
Apparently. I had expected that this would fail on Windows, but it just worked (I had previously used this action for `scipp-coverage`).

Reply:
Mambaforge uses `conda-forge` as the default channel and does not use the `anaconda` channel. Since 'main' action uses conda, there could be a discrepancy between the two builds. You used `--override-channels` with `conda-build`, so that is hopefully not an issue. So this is just a reminder for the future.